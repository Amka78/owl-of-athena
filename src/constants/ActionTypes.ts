const LOGIN = "LOGIN" as const;
const UPDATE_USER = "UPDATE_USER" as const;
const LOGOUT = "LOGOUT" as const;
const CACHE_SETTINGS = "CACHE_SETTINGS" as const;
const INIT = "INIT" as const;
const CACHE_PROFILES = "CACHE_PROFILES" as const;
const CACHE_SESSIONS = "CACHE_SESSIONS" as const;
const SELECT_SESSION = "SELECT_SESSION" as const;
const SELECT_SESSION_DETAIL = "SELECT_SESSION_DETAIL" as const;
const INITIALIZE_AURORA = "INITIALIZE_AURORA" as const;
const INITIALIZE_SESSION = "INITIALIZE_SESSION" as const;
const UPDATE_FILTER = "UPDATE_FILTER" as const;
const DELETE_SESSION = "DELETE_SESSION" as const;
const UPDATE_SESSION = "UPDATE_SESSION" as const;
export default {
    INIT,
    INITIALIZE_AURORA,
    INITIALIZE_SESSION,
    LOGIN,
    LOGOUT,
    UPDATE_USER,
    CACHE_SESSIONS,
    DELETE_SESSION,
    SELECT_SESSION,
    SELECT_SESSION_DETAIL,
    UPDATE_FILTER,
    UPDATE_SESSION,
    CACHE_SETTINGS,
    CACHE_PROFILES,
};
