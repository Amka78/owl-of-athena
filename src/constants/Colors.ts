const teal = "#00ffe6";
const purple = "#4a2fba";
const white = "#ffffff";
export default {
    black: "#000000",
    charcoal: "#333333",
    gray: "#9B9B9B",
    white: white,
    purple: purple,
    navy: "#070724",
    navy_darker: "#09092B",
    blue: "#001438",
    cyan: "#bcfeff",
    teal: teal,
    yellow: "#daf392",
    red: "#ff4310",
    light_orange: "#ff8260",
    semi_transparent: "#66000000",
    aurora_connected: teal,
    aurora_connected_text: purple,
    aurora_disconnected: purple,
    aurora_disconnected_text: white
};
