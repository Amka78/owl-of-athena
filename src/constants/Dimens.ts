export default {
    toolbar_title_text_size: 14,
    toolbar_button_text_size: 14,
    toolbar_button_margin_horizontal: 8,

    view_padding_bottom: 12,

    content_title_text_size: 24,
    content_title_margin_top: 32,
    content_title_margin_bottom: 48,
    content_text_size: 16,
    content_margin_horizontal: 48,

    snackbar_text_size: 8,

    statusbar_height: 8,

    navbar_height: 64,

    tip_text_size: 13,
    tip_text_margin_bottom: 13,

    error_text_size: 13,
    error_text_margin_bottom: 18,

    button_text_size: 24,
    button_height: 50,
    button_margin_bottom: 16,
    button_radius: 16,
    button_flat_text_size: 14,
    button_flat_margin_bottom: 12,
    button_flat_height: 28,

    input_text_size: 12,
    input_margin_bottom: 16,

    checkbox_text_size: 12,
    checkbox_outer_diameter: 24,
    checkbox_inner_diameter: 16,
    checkbox_gap: 4,
    checkbox_offset: 2,
    checkbox_padding_left: 16,
    checkbox_padding_right: 24,
    checkbox_margin_top: 16,
    checkbox_margin_bottom: 8,

    option_text_size: 14,
    option_height: 20,
    option_margin_bottom: 12,

    time_picker_height: 150,
    time_picker_scale: 1,
    time_picker_padding_horizontal: 0,

    circular_progress_diameter: 96,
    circular_progress_text_size: 32,

    // home fragment dimensions
    home_alarm_time_text_size: 48,
    home_alarm_meridian_text_size: 22,

    // session fragment dimensions
    session_label_text_size: 20,
    session_time_text_size: 28,
    session_sleep_score_text_size: 48,
    session_notes_text_size: 20,

    // session list fragment dimensions
    session_list_item_padding: 12,

    session_list_section_height: 24,
    session_list_section_padding_left: 24,
    session_list_section_text_size: 16,

    session_list_item_title_text_size: 20,
    session_list_item_subtitle_text_size: 15,
    session_list_item_text_margin_left: 18
};
