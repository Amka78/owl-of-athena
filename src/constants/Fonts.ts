export default {
    primaryRegular: "calibre_app_regular",
    primarySemiBold: "calibre_app_semibold"
};
