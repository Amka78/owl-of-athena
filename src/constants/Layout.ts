import { Dimensions } from "react-native";

const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

export default {
    isSmallDevice: width < 375,
    isLargeDevice: width > 1280,
    window: {
        height,
        width
    },
    maxWidth: 1280
};
