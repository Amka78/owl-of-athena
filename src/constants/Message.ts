import * as Localization from "expo-localization";
import i18n from "i18n-js";
import { MessageKeys } from "./";
import en from "./i18-n/en";
import ja from "./i18-n/ja";

i18n.fallbacks = true;
i18n.translations = { ja, en };
i18n.locale = Localization.locale;
i18n.defaultLocale = "ja-JP";
export type MessageKeys = typeof MessageKeys | string;
export type MessageLocalizationParam = {
    key: MessageKeys;
    restParam?: any[];
};
export default class Message {
    /**
     * 文言取得
     * @description 指定したキーに紐づく現在のロケールに対応する文言情報を取得します。{数字}のフィールドに文言の埋め込みが可能です。
     * @static
     * @memberof Message
     */
    public static get(value: MessageKeys, restParam?: MessageKeys[]): string {
        return Message.getTranslatedValue(value, restParam);
    }

    public static setLocale(localeStr: string): void {
        i18n.locale = localeStr;
    }

    private static getTranslatedValue(
        key: MessageKeys,
        restParam?: MessageKeys[]
    ): string {
        let currentLocaleValue = i18n.t(key.toString());

        if (currentLocaleValue.match("missing")) {
            currentLocaleValue = key as string;
        }
        if (restParam) {
            for (let i = 0; restParam.length > i; i++) {
                const settingValue = Message.get(restParam[i]);
                currentLocaleValue = currentLocaleValue.replace(
                    "{" + i + "}",
                    settingValue
                );
            }
        }
        return currentLocaleValue;
    }
}
