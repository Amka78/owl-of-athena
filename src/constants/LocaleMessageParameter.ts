import { MessageKeys } from "./Message";

export default class LocaleMessageParameter {
    constructor() {
        this.key = "";
    }
    public key: MessageKeys;
    public restParam?: MessageKeys[];
}
