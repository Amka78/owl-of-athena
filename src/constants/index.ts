import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
