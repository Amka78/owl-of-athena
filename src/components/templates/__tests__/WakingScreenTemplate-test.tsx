import "jest-enzyme";
import "react-native";

import { ShallowWrapper } from "enzyme";
import React from "react";

import { Message } from "../../../constants";
import {
    createMock,
    DesktopDimension,
    toJson,
} from "../../../utils/TestHelper";
import {
    WakingScreenTemplate,
    WakingScreenTemplateProps,
} from "../WakingScreenTemplate";

let component: ShallowWrapper<WakingScreenTemplateProps, unknown, unknown>;

describe("WakingScreenTemplate UnitTest", () => {
    it.each(["us", "ja"])("renders correctly", (locale: string) => {
        const testEvent: () => void = () => {
            return;
        };
        Message.setLocale(locale);
        component = createMock(
            <WakingScreenTemplate
                timeView={{ hours: 0, minutes: 0 }}
                onWakeupPress={testEvent}
                dimens={DesktopDimension}
            ></WakingScreenTemplate>
        );

        expect(toJson(component)).toMatchSnapshot();
    });
});
