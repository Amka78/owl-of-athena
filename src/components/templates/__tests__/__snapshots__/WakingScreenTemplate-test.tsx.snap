// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WakingScreenTemplate UnitTest renders correctly 1`] = `
<Component>
  <Component
    isDesktop={true}
  >
    Time to wake up!
  </Component>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Component
      hours={0}
      minutes={0}
      mode="meridian"
      timeMeridianStyle={
        Object {
          "fontSize": 22,
        }
      }
      timeStyle={
        Object {
          "fontSize": 48,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Component
      onPress={[Function]}
    >
      Wake Up
    </Component>
    <Component>
      Tip: You can use the Aurora’s button to stop the alarm.
    </Component>
  </View>
</Component>
`;

exports[`WakingScreenTemplate UnitTest renders correctly 2`] = `
<Component>
  <Component
    isDesktop={true}
  >
    起きてください!
  </Component>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Component
      hours={0}
      minutes={0}
      mode="meridian"
      timeMeridianStyle={
        Object {
          "fontSize": 22,
        }
      }
      timeStyle={
        Object {
          "fontSize": 48,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Component
      onPress={[Function]}
    >
      起きる
    </Component>
    <Component>
      Tip: アラームを止めるためにAuroraのボタンが使えます。
    </Component>
  </View>
</Component>
`;
